packages/swap/src/routes/swap.ts

```diff
@@ -130,9 +140,24 @@ router.get(
     );
 
     let state: State;
+    let error: { name: string; message: string } | undefined;
 
-    if (failedSwap) {
+    if (failedSwap || swap?.ignoredEgress) {
       state = State.Failed;
+      if (failedSwap?.reason) {
+        error = {
+          name: failedSwap.reason,
+          message: failedSwapMessage[failedSwap.reason],
+        };
+      } else if (swap?.ignoredEgress) {
+        const stateChainError = await prisma.stateChainError.findUniqueOrThrow({
+          where: { id: swap.ignoredEgress.stateChainErrorId },
```

Comment:
could we include this in the query above?

Reply:
You would think so, but it doesn't work